lib/Macros/Sources/SwiftMacros/TaskLocalMacro.swift

```diff
@@ -0,0 +1,204 @@
+    }
+
+    guard let name = firstBinding.pattern.as(IdentifierPatternSyntax.self)?.identifier else {
+      throw DiagnosticsError(
+        syntax: declaration,
+        message: "'@TaskLocal' property must have name", id: .incompatibleDecl)
+    }
+
+    let type = firstBinding.typeAnnotation?.type
+    let explicitType: String
+    if let type {
+      explicitType = ": TaskLocal<\(type.trimmed)>"
+    } else {
+      explicitType = ""
+    }
+
+    let initialValue: Any
+    if let initializerValue = firstBinding.initializer?.value {
+      initialValue = initializerValue
+    } else if let type, type.isOptional {
+      initialValue = "nil"
+    } else {
+      throw DiagnosticsError(
+        syntax: declaration,
+        message: "'@TaskLocal' property must have default value, or be optional", id: .mustBeVar)
+    }
+
+    // If the property is global, do not prefix the synthesised decl with 'static'
+    let isGlobal = context.lexicalContext.isEmpty
+    let staticKeyword: String
+    if isGlobal {
+      staticKeyword = ""
+    } else {
+      staticKeyword = "static "
+    }
+
+    return [
+      """
+      \(raw: staticKeyword)let $\(name)\(raw: explicitType) = TaskLocal(wrappedValue: \(raw: initialValue))
+      """
+    ]
+  }
+}
+
+extension TaskLocalMacro: AccessorMacro {
+  public static func expansion(
+    of node: AttributeSyntax,
+    providingAccessorsOf declaration: some DeclSyntaxProtocol,
+    in context: some MacroExpansionContext
+  ) throws -> [AccessorDeclSyntax] {
+    // We very specifically have to fail and diagnose in the accessor macro,
+    // rather than in the peer macro, since returning [] from the accessor
+    // macro adds another series of errors about it missing to emit a decl.
+    guard let varDecl = try requireVar(declaration) else {
+      return []
+    }
+    try requireStaticContext(varDecl, in: context)
+
+    guard let firstBinding = varDecl.bindings.first else {
+      return [] // TODO: make error
+    }
+
+    guard let name = firstBinding.pattern.as(IdentifierPatternSyntax.self)?.identifier else {
+      return [] // TODO: make error
+    }
+
+    return ["get { $\(name).get() }"]
+  }
+}
+
+@discardableResult
+private func requireVar(_ decl: some DeclSyntaxProtocol,
+                        diagnose: Bool = true) throws -> VariableDeclSyntax? {
+  if let varDecl = decl.as(VariableDeclSyntax.self) {
+    return varDecl
+  }
+  if diagnose {
+    throw DiagnosticsError(
+      syntax: decl,
+      message: "'@TaskLocal' can only be applied to properties", id: .mustBeVar)
+  }
+
+  return nil
+}
+
+@discardableResult
+private func requireStaticContext(_ decl: VariableDeclSyntax,
+                                  in context: some MacroExpansionContext,
+                                  diagnose: Bool = true) throws -> Bool {
+  let isStatic = decl.modifiers.contains { modifier in
+    modifier.name.text == "\(Keyword.static)"
```

Comment:
```suggestion
    modifier.kind == .keyword(.static)
```

should do the job without doing string comparisons.

Reply:
This specific one I thought didn't work hmmm, I'll double check